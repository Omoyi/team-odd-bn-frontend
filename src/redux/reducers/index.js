import { combineReducers } from 'redux';
import socialAuth from './socialAuthReducer';
import authReducers from './authReducers';
import profileReducers from './ProfileReducers';
import updateProfileReducer from './updateProfileReducer';
import getAllUsersReducer from './getAllUsersReducers';
import signoutReducer from './signoutReducer';
import tripReducers from './tripReducers';
import getAllCitiesReducers from './getAllCitiesReducers';
import accommodationReducer from './accommodation';
import multicityReducer from './multicityReducer';
import adminReducer from './admin/adminReducer';
import viewAccomodationReducer from './accommodation/viewAccomodationReducer';
import PaginationReducer from './PaginationReducer';
import searchReducer from './searchReducers/searchReducers';

import bookingReducer from './bookings';

export default combineReducers({
  viewProfile: profileReducers,
  updateProfile: updateProfileReducer,
  allUsers: getAllUsersReducer,
  socialAuth,
  auth: authReducers,
  signout: signoutReducer,
  trips: tripReducers,
  allCities: getAllCitiesReducers,
  accommodation: accommodationReducer,
  multicityTrip: multicityReducer,
  admin: adminReducer,
  allAccomodation: viewAccomodationReducer,
  pagination: PaginationReducer,
  bookings: bookingReducer,
  search: searchReducer,
});
