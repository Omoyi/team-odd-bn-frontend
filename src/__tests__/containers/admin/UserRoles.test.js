import React from 'react';
import { shallow, mount } from 'enzyme';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  UserRoles, mapStateToProps,
} from '../../../containers/admin/UserRoles';
import {
  defaultProps,
  updateDefaultProps,
  emptyDataComponent,
} from '../../../__mocks__/admin/admin';
import rootReducer from '../../../redux/reducers/index';

const middlewares = [thunk];
const testStore = (state) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, state);
};
const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = mount(
    <Router>

      <UserRoles {...defaultProps} store={store} />
    </Router>,
  );
  return wrapper;
};
const setUpEmptyTable = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(
    <Router>
      <UserRoles {...emptyDataComponent} store={store} />
    </Router>,

  );
  return wrapper;
};

describe('User roles test suite', () => {
  beforeEach(() => {
    const input = document.createElement('input');
    input.setAttribute('id', 'searchInput');
    document.body.appendChild(input);
  });
  beforeEach(() => {
    const div = document.createElement('div');
    div.innerHTML = `
    <div class="entity">
    <div class="firstName">user</div>
    <div class="lastName">test</div>
    <div class="Email">me@you.com</div>
    <div class="role">user</div>
    </div>
    <div class="entity">
    <div class="firstName">user</div>
    <div class="lastName">test</div>
    <div class="Email">admin@gmail.com</div>
    <div class="role">user</div>
    </div>
    `;
    div.setAttribute('class', 'tableContainer');
    document.body.appendChild(div);
  });

  it('Should mount without table data', () => {
    const component = setUpEmptyTable({});
    const componentDidMountSpy = jest.spyOn(
      component.instance(),
      'componentDidMount',
    );
    component.find('[data-test="userRoles-test"]');
    expect(componentDidMountSpy).not.toHaveBeenCalled();
  });
  it('Should call componentDidMount Successfully', () => {
    const component = setUp({});
    const componentDidMountSpy = jest.spyOn(component.instance(), 'componentDidMount');
    component.find('[data-test="userRoles-test"]');
    expect(componentDidMountSpy).not.toHaveBeenCalled();
  });
  it('Should be able to search', () => {
    const component = setUp({});
    component.find('[data-test="dataFilter-test"]').simulate('change', 'admin');
    component.find('[data-test="dataFilter-test"]').simulate('change', 'me');
  });
  it('should map state to props', () => {
    const initialState = {};
    expect(mapStateToProps(initialState).adminState).toBeDefined();
  });
});
