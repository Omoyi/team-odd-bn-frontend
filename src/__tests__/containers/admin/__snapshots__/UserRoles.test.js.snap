// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render should render successfully 1`] = `ShallowWrapper {}`;
